### PR TITLE
styles(search): implement and document variables and attributes

### PR DESCRIPTION
**Pull request by badejayayesubabu**
## Description

`<hx-search>` : implement and document variables and attributes

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2125

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/scss/components/search/SearchFacade.scss [hunk 1]
```diff
@@ -6,30 +6,38 @@
 
 @mixin SearchFacadePristineFocus {
   border: 1px solid $cyan-700;
```
Comment (100stacks) [inline src/scss/components/search/SearchFacade.scss:8]: Please use updated  `v2.0` colors to replace `$cyan-XXX`.

File: src/scss/components/search/SearchFacade.scss [hunk 2]
```diff
@@ -6,30 +6,38 @@
...
+  border: 1px solid var(--hxSearch-SearchFacadePristineFocus-border, $cyan-700);
```
Comment (100stacks) [inline src/scss/components/search/SearchFacade.scss:9]: Please use updated  `v2.0` colors to replace `$cyan-XXX`.

File: src/scss/components/search/_index.scss [hunk 3]
```diff
@@ -61,6 +61,7 @@ hx-search-control {
   > input[type="search"]:focus ~ hx-search,
   > button.hxClear:focus ~ hx-search {
     border-color: $cyan-700;
```
Comment (100stacks) [inline src/scss/components/search/_index.scss:63]: Please use updated  `v2.0` colors to replace `$cyan-XXX`.

File: src/scss/components/search/_index.scss [hunk 4]
```diff
@@ -61,6 +61,7 @@ hx-search-control {
...
+    border-color: var(--hxSearch-border-color, $cyan-700);
```
Comment (100stacks) [inline src/scss/components/search/_index.scss:64]: Please use updated  `v2.0` colors to replace `$cyan-XXX`.

File: src/scss/components/search/_index.scss [hunk 5]
```diff
@@ -223,5 +235,6 @@ hx-search-control:not([hx-defined]) > button {
 
   &:hover {
     background-color: $cyan-50;
```
Comment (100stacks) [inline src/scss/components/search/_index.scss:237]: Please use updated  `v2.0` colors to replace `$cyan-XXX`.

File: src/scss/components/search/_index.scss [hunk 6]
```diff
@@ -223,5 +235,6 @@ hx-search-control:not([hx-defined]) > button {
...
+    background-color: var(--hxSearch-hxSearchSuggestion-hover-bgcolor, $cyan-50);
```
Comment (100stacks) [inline src/scss/components/search/_index.scss:238]: Please use updated  `v2.0` colors to replace `$cyan-XXX`.

**Review by 100stacks — CHANGES_REQUESTED** [review]
### Please use updated  `v2.0` colors variables.

File: src/scss/components/search/SearchFacade.scss (hunk 1 above)
Comment (badejayayesubabu) [reply src/scss/components/search/SearchFacade.scss]: Changes have been done, Please review

File: src/scss/components/search/_index.scss (hunk 3 above)
Comment (badejayayesubabu) [reply src/scss/components/search/_index.scss]: Changes have been done, Please review

File: src/scss/components/search/_index.scss (hunk 4 above)
Comment (badejayayesubabu) [reply src/scss/components/search/_index.scss]: Changes have been done, Please review

File: src/scss/components/search/_index.scss (hunk 5 above)
Comment (badejayayesubabu) [reply src/scss/components/search/_index.scss]: Changes have been done, Please review

File: src/scss/components/search/_index.scss (hunk 6 above)
Comment (badejayayesubabu) [reply src/scss/components/search/_index.scss]: Changes have been done, Please review

**Review by 100stacks — APPROVED** [review]
Good job! Dev LGTM

**Pull request merged**